apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/blockchain/BlockchainImpl.java

```diff
@@ -298,7 +301,52 @@ public Block loadBlock(Connection con, ResultSet rs, boolean loadTransactions) {
     @Override
     public void saveBlock(Connection con, Block block) {
         blockDao.saveBlock(con, block);
-        transactionDao.saveTransactions(con, block.getOrLoadTransactions());
+        transactionDao.saveTransactions(con, this.getOrLoadTransactions(block));
+    }
+
+    public List<Transaction> getOrLoadTransactions(Block parentBlock) {
+        if (parentBlock.getTransactions() == null || parentBlock.getTransactions().size() == 0) {
+            List<Transaction> blockTransactions = this.getBlockTransactions(parentBlock.getId());
+            if (blockTransactions.size() > 0) {
+                List<Transaction> transactions = Collections.unmodifiableList(blockTransactions);
+                short index = 0;
+                for (Transaction transaction : transactions) {
+                    transaction.setBlock(parentBlock);
+                    transaction.setIndex(index++);
+                    ((TransactionImpl) transaction).bytes();
+                    transaction.getAppendages();
+                }
+                parentBlock.setTransactions(transactions);
+            } else {
+                parentBlock.setTransactions(Collections.emptyList());
+            }
+        } else if (parentBlock.getTransactions() == null) {
+            parentBlock.setTransactions(Collections.emptyList());
+        }
+        return parentBlock.getTransactions();
+    }
+
+    public JSONObject getJSONObject(Block block) {
+        JSONObject json = new JSONObject();
```

Comment:
I'd prefer to create separate BlockJsonSerializer for such code snippet

Reply:
Code is moved to another component